### DISCUSSION TITLE
Help with the relationship model User

### DISCUSSION THREAD

**Discussion by HENRYHKll** [fastapi-users/fastapi-users#1421] [Q&A] [answered]
here is my code
https://github.com/HENRYHKll/fastapi-users-sa-relationship
migrations are completed successfully, and FastApi is also running successfully. But if I make a request, for example /auth/jwt/login an error message appears `sqlalchemy.exc.InvalidRequestError: When initializing mapper Mapper[User(user)], expression 'Profile' failed to locate a name ('Profile'). If this is a class name, consider adding this relationship() to the <class 'user.models.User'> class after both dependent classes have been defined.`

**Comment by frankie567** [comment]
From what I see, you set the `back_populates` option on `User`'s side to `profile` (lowercase). On the `Profile` model side, you declared that property as `Profile` (with a capital *P*). Try to rename it to `profile`, all lowercase.

**Reply by HENRYHKll** [reply, marked as answer]
thanks for the answer! 
I solved the problem by adding a backref
https://github.com/HENRYHKll/fastapi-users-sa-relationship/blob/ab2f4b344d8a17343e5bba5b01353e7b5b2d1470/profile/models.py#L23
and in order for it to work on the asynchronous engine, it was necessary to add AsyncAttrs to the Bdse model
and call through awaitable_attrs
https://github.com/HENRYHKll/fastapi-users-sa-relationship/blob/ab2f4b344d8a17343e5bba5b01353e7b5b2d1470/profile/crud.py#L21
